packages/contacts/services/request.js

```diff
@@ -91,6 +92,46 @@ module.exports = {
           return;
         }
 
+        // Check, if an invite capability URI is present. If so, accept the request automatically.
+        if (activity['as:context']) {
+          const capability = await ctx.call('auth.getValidateCapability', {
+            capabilityUri: activity['as:context'],
+            webId: recipientUri
+          });
+          const { url: profileUri } = await ctx.call('ldp.resource.get', {
+            resourceUri: recipientUri,
+            webId: 'system',
+            accept: MIME_TYPES.JSON
+          });
+          // Capability to read the profile implies automatic contact approval.
+          if (
+            capability.type === 'acl:Authorization' &&
+            capability['acl:Mode'] === 'acl:Read' &&
+            capability['acl:AccessTo'] === profileUri
+          ) {
+            // Send the accept.
+            await ctx.call('activitypub.outbox.post', {
+              collectionUri: recipient.outbox,
+              type: ACTIVITY_TYPES.ACCEPT,
+              actor: recipient.id,
+              object: activity.id,
+              to: activity.actor
+            });
+            // Send notification.
+            /*
+            // TODO: i18n & fetch user data.
+            await ctx.call('activitypub.outbox.post', {
+              collectionUri: recipient.outbox,
+              type: ACTIVITY_TYPES.NOTE,
+              actor: recipient.id, // TODO: Actually, the application itself.
+              summary: `Contact request by ${recipient.id} accepted.`,
+              content: `The user ${recipient.id} was accepted to your contacts since they had your invite URI.`,
+              to: recipient.id
+            });
+            */
```

Comment:
I proceeded by modifying https://github.com/assemblee-virtuelle/semapps/blob/dfc0f2df719bf3f2f7897bbb5b9347c683065ff2/src/middleware/packages/activitypub/utils/matchActivity.js#L11 to support the pattern to be a function.

Since your referenced code of the ActivitiesHandlerMixin proceeds similarly, we might want to refactor that at some point. But I wasn't sure, if I might crash something by migrating the custom logic there to the new code and taking a close look might cost quite some time. Maybe you have a better overview of implications :)
I'll create an issue.